_pages/release_notes.md

```diff
@@ -8,7 +9,121 @@ image:
+
+Enterprise customers will see the feature enabled with their quarterly update. All necessary configurations are documented in the Azure Listeners section in our dev center.
+
+The feature is in the main algorithm dashboard under tab event listeners, and listeners can be created from the “Create New” dropdown menu at top right. An event listener is created for a published version of the algorithm. The listener is listening to a specified queue and whenever a message is received from the queue, the listener executes the algorithm with the message payload as the input.
+
+### Algorithm environments ###
+
+We have improved the way algorithm runtime environments are built, tested, and provided to users, by creating a new system to deliver environments and languages that can be executed within an Algorithmia cluster; we’ve ensured that these environments can be executed in any enterprise cluster. 
+
+#### Using the feature ####
+
+##### For admins #####
+
+Algorithm Environments are maintained by Algorithmia, and can be imported into a given cluster through the admin panel. Cluster administrators can choose to import any environment available in the list to their cluster based on the needs of their users and security concerns for the operator. 
+
+Admins will see a new page for Algorithm Environments, which will allow them to import these new languages. 
+
+Once logged into Algorithmia, navigate to the Admin Panel and click on “Add Environment” to see the list of available environments; then click on the desired environment(s) to add, and then click “Add Environment” to equip it.
+
+By selecting and adding the environment, your users will be able to create algorithms using that installed environment. If you have not added the environment from the list, your users will not be able to view that environment or create an algorithm using that environment.
+
+The specified environment will now be available in the list of environments, and details can be viewed by clicking on a specific one.
+
+##### Non admins #####
+
+When making new algorithms, users will see language options that are now called Environments. 
+
+### Improved debugging of algorithms (local and UI) ###
+
+This feature improves users’ local debugging experience by updating error messages, enabling users to write and run local tests for cloned algorithms, and passing in local data files as input to the algorithms. Our new integration with PyCharm means users can also use an effective and familiar debugging tool for algorithm development.
+
+Enabling users to pass in their local data file into the Python Client will make it faster to test algorithms locally before production.
+
+#### Using the feature ####
+
+Set up a local development environment, including installing the Algorithmia Python Client in PyCharm. Full directions for integrating Algorithmia with PyCharm is in the Developer Center under Advanced Algorithm Development.
```

Comment:
This page was deployed yesterday, so it's safe to link to now if you want:
```suggestion
Set up a local development environment, including installing the Algorithmia Python Client in PyCharm. Full directions for integrating Algorithmia with PyCharm is in the Developer Center [under Advanced Algorithm Development](/developers/algorithm-development/advanced-algorithm-development/pycharm).
```

Reply:
Added!